### ISSUE TITLE
Feature Request: Support for Per-Request Logits Post-Processor Registration

### ISSUE THREAD

**Issue by EmileDqy** [NVIDIA/TensorRT-LLM#2809]
## Issue

Currently, logits post-processors must be registered at initialization time through `ExecutorConfig.logits_post_processor_map`. This prevents runtime registration of post-processors and schemas on a per-request basis.

```cpp
using LogitsPostProcessor = std::function<void(IdType, Tensor&, BeamTokens const&, StreamPtr const&, std::optional<IdType>)>;
using LogitsPostProcessorMap = std::unordered_map<std::string, LogitsPostProcessor>;
```

> I noticed `Disaggregated Serving` in the [2025 Roadmap](https://docs.google.com/presentation/d/1gycPmtdh7uUcH6laOvW65Dbp9F1McUkGDIcAyjicBZs/edit#slide=id.g31912614093_0_3), but I am unsure of its scope and whether it is related to this issue.

## Impact

Using Triton Inference Server with TensorRT-LLM backend in production:

- Tight coupling between application and model deployment
- The full set of validation schemas must be known and named at model build time
- Application logic changes require model redeployment
- Inference server cannot be schema-agnostic

## Proposed Solution

Add support for single-use logits post-processors scoped to individual requests. The post-processor would be registered for the duration of the request only and automatically cleaned up afterwards, following the pattern established by vLLM and TGI's grammar implementations for constrained decoding. This functionality can be used in addition to the `LogitsPostProcessorMap` declared at initialization time, ensuring backward compatibility.

> TGI has recently integrated TensorRT-LLM but [had to disable its grammar support due to this limitation](https://github.com/huggingface/text-generation-inference/blob/06dfe9abfe958d87a1f4aa1fe960c765e2deb900/backends/trtllm/src/looper.rs#L296-L298). This suggests broader ecosystem benefits.

## Example Use Case

Using LM Format Enforcer with dynamic schemas that depend on request context. The current workaround requires pre-registering all possible schema combinations at initialization time, which can grow exponentially with context complexity.

**Comment by achartier** [comment]
Support has been added in the last upgrade, see https://github.com/NVIDIA/TensorRT-LLM/blob/77d7fe1eb2d89753ae33a0745d1083c133064f92/cpp/include/tensorrt_llm/executor/executor.h#L647

**Comment by EmileDqy** [comment]
Hello,
Thanks a lot for the reply and for the upgrade! Our team will try this feature through the Tensorrt-LLM backend for triton very soon. 
I'm closing the issue now.

Have a nice day
